docs/slack-spring-boot-docs/src/main/asciidoc/dev-features/kotlin-sample-extensions.adoc

```diff
@@ -1,11 +1,152 @@
+
+This is how the first method will look like:
+
+[source,kotlin]
+--
+include::../../../../../../samples/slack-spring-boot-starter-sample/src/test/kotlin/com/kreait/slack/sample/ResponseCommandReceiverTest.kt[lines=17..34]
+--
+
+Our ```mockSlackClient``` invokes the .respond().message() methods with a payload ("www.test.com") and sets the successResponse value to ```Unit``` since we don't return anything upon successfully posting a response.
+By setting ```successResponse``` we also state that the invocation is treated as "successful".
+
+We then mock the https://github.com/kreait/slack-spring-boot-starter/blob/master/samples/slack-spring-boot-starter-sample/src/main/kotlin/com/kreait/slack/sample/ResponseHandler.kt[ResponseHandler] so we can post a message in Slack with the status of the invocation in either the ```.onSuccess``` or ```.onFailure``` block.
+
+The declaration of the ResponseCommandReceiver is provided with our mockSlackClient and the responseHandler so the actual test can be done.
+
+We call the ```onReceiveSlashCommand``` from our just declared commandReceiver and provide it with the neccessary parameters.
+The first parameter is a sample of a SlackCommand - but here it is completely empty.
+In combination with the copy-Method we get a duplicate of this sample but can pick the parameters we like to alternate without having to fill in every parameter again.
+In this example we change the ```command```, the ```channelId``` and the ```responseUrl```.
+Additionally we need to provide HttpHeaders and a Token.
+
+In this case HTTPHeaders are Empty and the Token is received by the Team in the https://github.com/kreait/slack-spring-boot-starter/blob/master/starter/slack-spring-boot/src/main/kotlin/com/kreait/slack/broker/store/team/TeamStore.kt[TeamStore] we provided.
+
+[NOTE]
+====
+The TeamStore is used to persist teams after the installation of a Slack App (either file or inMemory) since you'd have to reinstall the app after every start which in fact would be annoying.
+The starter provides the inMemory-Store on default.
+_This is not thought to be used in a production environment._
+====
+
+As our test method is used to verify a success, we first call the receiver:
+
+[source,kotlin]
+--
+include::../../../../../../samples/slack-spring-boot-starter-sample/src/test/kotlin/com/kreait/slack/sample/ResponseCommandReceiverTest.kt[lines=28..30]
+--
+
+After calling the ```onReceiveSlashCommand``` method we want to verify if the right ```ResponseHandler``` method was at least invoked once.
+
+[NOTE]
+====
+Remember that the methods provided by the ResponseHandler are used in the .onSuccess or .onFailure blocks so we want to verify *their* usage and not the usage of the actual onReceiveSlashCommand method.
+====
+
+We verify if successResponse is used once and *after* that we verify that the failureResponse is used zero times.
```

Comment:
I would say: 
We verify that the successResponse-method is invoked once and the failureResponse-method is invoked zero times